microsoft/testsuites/core/msr.py

```diff
@@ -116,16 +122,42 @@ def verify_hyperv_platform_id(self, node: RemoteNode) -> None:
         else:
             raise SkippedException("MSR platform id test not yet supported on this OS.")
 
-        distro.install_packages("msr-tools")
-        node.tools[Modprobe].load("msr")
+        # get the msr offset to read, this constant is arch specific
+        arch_id = node.tools[Lscpu].get_architecture()
+        try:
+            arch_msr_offset = HvOsPlatformInfo.HV_REGISTER_GUEST_OSID[arch_id]
+        except KeyError as missing_key:
+            raise SkippedException(f"Arch {missing_key} is not supported by msr test")
+
+        # try installing msr-tools if rdmsr isn't already insalled.
+        if node.execute("command -v rdmsr", shell=True, sudo=True).exit_code != 0:
+            try:
+                distro.install_packages("msr-tools")
+            except AssertionError:
+                raise SkippedException(
```

Comment:
for redhat rhel-arm64 8_6-arm64 8.6.2022060901 image, case fails for below exception. it is not AssertionError type
```failed. MissingPackagesException: Package manager could not install packages: msr-tools```

Reply:
Nice, will fix. I guess this is a bug for the other distros? There doesn't seem to be a common failure mode for installing packages. Some return without installing, some throw an AssertionError, some throw a MissingPackagesException (I think this is the correct failure mode)

Reply:
fixed :)